samples/host_exerciser/host_exerciser_cmd.h

```diff
@@ -36,26 +36,48 @@ namespace host_exerciser {
 
 class fpgaperf {
 public:
-  fpga_result perfget(token::ptr_t token, fpga_perf_counter *fpga_perf)
+  static std::shared_ptr<fpgaperf> get(token::ptr_t token)
   {
-      return fpgaPerfCounterGet(token->c_type(), fpga_perf);
+    /*std::make_shared function doesn’t have access to the private constructor
+     *of fpgaperf, and as such it cannot instantiate it. Created temporary
+     *class that inherits from fpgaperf.*/
+    struct shared_enabler : public fpgaperf {};
+    auto p = std::make_shared<shared_enabler>();
+    if (fpgaPerfCounterGet(token->c_type(), p->counter_) != FPGA_OK) {
+        std::cout << "Failed to get the fpga perf counter" << std::endl;
+        p.reset();
+        return NULL;
+    }
+    return p;
```

Comment:
I think this is overcomplicating things a bit.
The example I gave was mostly for reference and so I hadn't tested it.
If you're getting compiler errors with make_shared, then I suggest using something simpler:
```c++
  std::shared_ptr<fpgaperf> p(new fpgaperf());
  if (fpgaPerfCounterGet(token->c_type(), p->counter_) != FPGA_OK) {
    p.reset();
  }
  return p;

```